### DISCUSSION TITLE
It doesn’t work on Alpine Linux, neither with Libvirt nor with Docker.

### DISCUSSION THREAD

**Discussion by Rosnerv** [winapps-org/winapps#351] [Support]
Hello,

I’ve been trying to get Winapps working on Alpine Linux, but no matter what I do, it doesn’t connect to the virtual machine with Libvirt or the Docker container. I believe I have all the dependencies installed; however, since it’s a distro that uses Musl instead of Glibc, I’m not sure if any additional configuration is needed. The error when running the Winapps installer is always the same: it can’t find the virtual machine in Libvirt, it can’t find Windows, or it can’t connect to the container with FreeRDP.

Can you please help me?

I use ChatGPT for translate.

**Comment by Rosnerv** [comment]
Here, the help stinks; it's unbelievable that no one says anything, neither those who maintain this nor anyone else.

That's why many lose motivation and leave Linux, because help is scarce, and even with ChatGPT now, it was worse before...

**Comment by oskardotglobal** [comment]
Let me rephrase that for you: "Thanks for spending your time to write and maintain free software for anyone to use and trying to support users and fix bugs, even though people open no-effort issues along the lines of 'it doesnt work pls fix'."

Maybe ChatGPT can write you a better piece of software? We're always interested in new opportunities to enhance WinApps. Feel free to open a pull request.

**Comment by Rosnerv** [comment]
> Permíteme reformular eso para ti: "Gracias por dedicar tu tiempo a escribir y mantener software libre para que cualquiera pueda usarlo y tratar de ayudar a los usuarios y corregir errores, incluso cuando la gente abre problemas sin esfuerzo del tipo 'no funciona, por favor arréglalo'".
> 
> ¿Quizás ChatGPT pueda escribirte un software mejor? Siempre nos interesan nuevas oportunidades para mejorar WinApps. No dudes en abrir una solicitud de incorporación de cambios.


Hello friend, I believe you don't understand the point. I have been trying everything to make WinApps work on Alpine Linux KDE Plasma 3.20, and I haven't achieved anything. That's why I'm turning to official support, which I believe is yours. However, I can't find any information or support, and that's why I'm frustrated.

In all other distros that use Systemd, there are no problems; I have even gotten WinApps to work on Void Linux, which uses runit. However, in the distro that is most commonly used for Docker containers, I haven't been able to. I don't know if it's a problem with the distro, if something is missing, or if WinApps is not compatible with Alpine because it uses Musl instead of Glibc and OpenRC instead of Systemd. Honestly, I don't know.

What I do know is that I want it to work, and I would appreciate any help.

**Comment by oskardotglobal** [comment]
Please share more information about your problem (logs, screenshots, ...)

**Comment by oskardotglobal** [comment]
You will have to build FreeRDP3 from source though, it seems like even the alpine edge repositories only contain FreeRDP2

**Comment by Rosnerv** [comment]
> Por favor, comparte más información sobre tu problema (registros, capturas de pantalla, ...)

[05:52:17:077] [14868:00003a17] [WARN][com.freerdp.crypto] - [verify_cb]: Certificate verification failure 'self-signed certificate (18)' at stack position 0
[05:52:17:077] [14868:00003a17] [WARN][com.freerdp.crypto] - [verify_cb]: CN = RDPWindows
[05:52:17:079] [14868:00003a17] [ERROR][com.winpr.sspi.Kerberos] - [kerberos_AcquireCredentialsHandleA]: krb5glue_get_init_creds (unable to reach any KDC in realm LOCALDOMAIN [-1765328228])
[05:52:17:079] [14868:00003a17] [ERROR][com.winpr.sspi.Kerberos] - [kerberos_AcquireCredentialsHandleA]: krb5glue_get_init_creds (unable to reach any KDC in realm LOCALDOMAIN [-1765328228])
[05:52:17:109] [14868:00003a17] [WARN][com.freerdp.core.license] - [license_read_binary_blob_data]: license binary blob::type BB_ERROR_BLOB, length=0, skipping.
[05:52:17:229] [14868:00003a17] [INFO][com.freerdp.gdi] - [gdi_init_ex]: Local framebuffer format  PIXEL_FORMAT_BGRA32
[05:52:17:229] [14868:00003a17] [INFO][com.freerdp.gdi] - [gdi_init_ex]: Remote framebuffer format PIXEL_FORMAT_BGRA32
[05:52:17:233] [14868:00003a4b] [INFO][com.freerdp.channels.rdpdr.client] - [devman_load_device_service]: Loading device service drive [home] (static)
[05:52:17:233] [14868:00003a17] [INFO][com.freerdp.channels.rdpsnd.client] - [rdpsnd_load_device_plugin]: [static] Loaded alsa backend for rdpsnd
[05:52:17:233] [14868:00003a17] [INFO][com.freerdp.channels.drdynvc.client] - [dvcman_load_addin]: Loading Dynamic Virtual Channel rdpsnd
[05:52:17:233] [14868:00003a17] [INFO][com.freerdp.channels.drdynvc.client] - [dvcman_load_addin]: Loading Dynamic Virtual Channel audin
[05:52:17:233] [14868:00003a17] [INFO][com.freerdp.channels.audin.client] - [audin_load_device_plugin]: Loaded oss backend for audin
[05:52:17:233] [14868:00003a17] [INFO][com.freerdp.channels.drdynvc.client] - [dvcman_load_addin]: Loading Dynamic Virtual Channel ainput
[05:52:17:233] [14868:00003a17] [INFO][com.freerdp.channels.drdynvc.client] - [dvcman_load_addin]: [05:52:17:233] [14868:00003a17] [WARN][com.winpr.thread] - [SetThreadPriority]: pthread_setschedprio(0) not implemented, requires POSIX 2008 or later
[05:52:18:859] [14868:00003a17] [WARN][com.freerdp.client.x11] - [xf_Pointer_get_window]: xf_Pointer: Invalid appWindow
[05:52:18:859] [14868:00003a17] [WARN][com.freerdp.client.x11] - [xf_Pointer_Set]: handle=0
[05:52:18:982] [14868:00003a17] [WARN][com.freerdp.client.x11] - [xf_Pointer_get_window]: xf_Pointer: Invalid appWindow
[05:52:18:982] [14868:00003a17] [WARN][com.freerdp.client.x11] - [xf_Pointer_Set]: handle=0
[05:52:18:401] [14868:00003a50] [WARN][com.freerdp.channels.drdynvc.client] - [check_open_close_receive]: {Microsoft::Windows::RDS::DisplayControl:11} OnOpen=0, OnClose=0x7f80a8284da0
[05:52:19:161] [14868:00003a17] [WARN][com.freerdp.client.x11] - [xf_Pointer_get_window]: xf_Pointer: Invalid appWindow
[05:52:19:161] [14868:00003a17] [WARN][com.freerdp.client.x11] - [xf_Pointer_Set]: handle=0
[05:52:19:367] [14868:00003a17] [ERROR][com.freerdp.core] - [rdp_set_error_info]: ERRINFO_LOGOFF_BY_USER [0x0001000C]
[05:52:19:377] [14868:00003a50] [WARN][com.freerdp.channels.drdynvc.client] - [check_open_close_receive]: {Microsoft::Windows::RDS::DisplayControl:11} OnOpen=0, OnClose=0x7f80a8284da0
Loading Dynamic Virtual Channel rdpgfx
[05:52:17:233] [14868:00003a17] [INFO][com.freerdp.channels.drdynvc.client] - [dvcman_load_addin]: Loading Dynamic Virtual Channel disp
[05:52:18:539] [14868:00003a50] [INFO][com.freerdp.channels.rdpsnd.client] - [rdpsnd_load_device_plugin]: [dynamic] Loaded alsa backend for rdpsnd
[05:52:18:336] [14868:00003a50] [INFO][com.freerdp.channels.rdpsnd.client] - [rdpsnd_load_device_plugin]: [dynamic] Loaded alsa backend for rdpsnd
[05:52:19:453] [14868:00003a4b] [INFO][com.freerdp.channels.rdpdr.client] - [device_announce]: registered [    drive] device #1: home (type=8 id=1)
[05:52:19:367] [14868:00003a17] [INFO][com.freerdp.core] - [rdp_print_errinfo]: ERRINFO_LOGOFF_BY_USER (0x0000000C):The disconnection was initiated by the user logging off their session on the server.

localhost:~/winapps$ ./setup.sh
################################################################################
#                                                                              #
#                            WinApps Install Wizard                            #
#                                                                              #
################################################################################

Q) Install or uninstall WinApps? --> Install
Q) Configure WinApps for the current user 'rosner' or the whole system? --> Current User
Q) Automatically install supported applications or choose manually? --> Automatic

WinApps installation already present at /home/rosner/.local/bin/winapps-src. Updating...
Already up to date.
Installing WinApps.
Checking for existing conflicting WinApps installations... Done!
Attempting to load WinApps configuration file... Done!
Checking whether dependencies are installed... Done!
Checking whether the user 'rosner' is part of the required groups... Done!
Checking the status of the Windows VM... Done!
Checking for an open RDP Port on Windows... Done!
Attempting to establish a Remote Desktop connection with Windows... Failed!

ERROR: REMOTE DESKTOP PROTOCOL FAILURE.
FreeRDP failed to establish a connection with Windows.
--------------------------------------------------------------------------------
Please view the log at /home/rosner/.local/share/winapps/FreeRDP_Test_20241022_0552_.log.
Troubleshooting Tips:
  - Ensure the user is logged out of Windows prior to initiating the WinApps installation.
  - Ensure the credentials within the WinApps configuration file are correct.
  - Utilise a new certificate by removing relevant certificate(s) in /home/rosner/.config/freerdp/server.
  - If using 'libvirt', ensure the Windows VM is correctly named as specified within the README.
  - If using 'libvirt', ensure 'Remote Desktop' is enabled within the Windows VM.
  - If using 'libvirt', ensure you have merged 'RDPApps.reg' into the Windows VM's registry.
  - If using 'libvirt', try logging into and back out of the Windows VM within 'virt-manager' prior to initiating the WinApps installation.
--------------------------------------------------------------------------------
Exiting with status '14'.
localhost:~/winapps$ 
![image](https://github.com/user-attachments/assets/18eb5d2e-84c8-4d6d-a1e3-e58905125fc3)
![image](https://github.com/user-attachments/assets/0d157936-3813-483e-bc64-bab04b6a0f7b)

**Comment by Rosnerv** [comment]
It only doesn't connect with Winapps; when connecting directly with FreeRDP, it does connect.

![image](https://github.com/user-attachments/assets/8d5220b1-d278-4ed9-a228-caad17f16921)

**Comment by oskardotglobal** [comment]
This seems to be the same issue as #270

**Comment by Rosnerv** [comment]
> Este parece ser el mismo problema que [el número 270](https://github.com/winapps-org/winapps/issues/270)

Unfortunately, none of the proposed solutions in those issues resolve the error.

**Comment by Rosnerv** [comment]
> Este parece ser el mismo problema que [el número 270](https://github.com/winapps-org/winapps/issues/270)

The same happens with the Docker version.

![image](https://github.com/user-attachments/assets/25b1313f-58e7-4075-bd53-c93056f1b33c)
![image](https://github.com/user-attachments/assets/be4c9ea8-0e65-498e-aba8-96afd5c3102c)

**Comment by semedilino-073** [comment]
For anyone still wondering, this was the fix for me:
https://gist.github.com/TheEEs/ebae8423a4b4fae7cb9531830b62d186
It is a customized version of setup.sh to run on Alpine systems
